### PR TITLE
API more changes to support UI my registrations table

### PR DESCRIPTION
**Pull request by doug-lovett**
Signed-off-by: Doug Lovett <doug@diamante.ca>

*Issue #:* /bcgov/entity#7852

*Description of changes:*
- Add registering name column. Modify data model add drafts.user_id.
- Add get registrations list "collapse" query parameter. Collapse if true returns a hierarchical parent/child representation of financing statements and registrations.
- Add get individual financing statement "current" query parameter. Current if true returns the current view of the financing statement.
- Report add legacy "RC" registry correction change type to report change type descriptions.
- Improve performance of report data generation: do not use pycountry to get CA, US descriptions.
- Improve performance of debtor individual search: always use indexed columns.     

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-api/src/ppr_api/models/draft.py [hunk 1]
```diff
@@ -58,6 +58,7 @@ class DraftTypes(Enum):
     draft = db.Column('draft', db.JSON, nullable=False)
     registration_number = db.Column('registration_number', db.String(10), nullable=True)
     update_ts = db.Column('update_ts', db.DateTime, nullable=True)
+    user_id = db.Column('user_id', db.String(1000), nullable=True)
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/models/draft.py:61]: should this be a foreign key to make it easier to link?

File: ppr-api/src/ppr_api/resources/drafts.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 from http import HTTPStatus
 
-from flask import request, jsonify
+from flask import request, jsonify, g
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/resources/drafts.py]: doesn't the linter complain about alphabetical order here?

**Review by thorwolpert — COMMENTED** [review]
couple of small things

File: ppr-api/src/ppr_api/resources/drafts.py (hunk 2 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/resources/drafts.py]: No complaints but sorted anyway as a second commit.

File: ppr-api/src/ppr_api/models/draft.py (hunk 1 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/models/draft.py:61]: I considered it, but the direct api consumers may submit requests with no users record, throwing a foreign constraint violation. With a constraint, we need to check/insert every time on high volume get drafts/get financing statements/get registrations requests. So I left the constraint out and use subqueries which allow for no users record.

**Pull request merged**